libraries/chain/combined_database.cpp

```diff
@@ -101,18 +101,25 @@ namespace eosio { namespace chain {
          kv_database(std::make_unique<b1::chain_kv::database>(rocksdb_path.c_str(), rocksdb_create_if_missing, rocksdb_threads, rocksdb_max_open_files)),
          kv_undo_stack(std::make_unique<b1::chain_kv::undo_stack>(*kv_database, rocksdb_undo_prefix)) {}
 
-   void combined_database::set_backing_store(backing_store_type backing_store) {
-      if (backing_store == backing_store_type::ROCKSDB) {
-         if (db.get<kv_db_config_object>().backing_store == backing_store_type::ROCKSDB)
-            return;
-         auto& idx = db.get_index<kv_index, by_kv_key>();
-         auto  it  = idx.lower_bound(boost::make_tuple(name{}, std::string_view{}));
-         EOS_ASSERT(it == idx.end(), database_move_kv_disk_exception,
-                    "Chainbase already contains KV entries; use resync, replay, or snapshot to move these to "
-                    "rocksdb");
-         db.modify(db.get<kv_db_config_object>(), [](auto& cfg) { cfg.backing_store = backing_store_type::ROCKSDB; });
+   void combined_database::check_backing_store_setting() {
+      switch (backing_store) {
+      case backing_store_type::CHAINBASE:
+         EOS_ASSERT(db.get<kv_db_config_object>().backing_store != backing_store_type::ROCKSDB, database_move_kv_disk_exception,
+                    "Chainbase indicates that RocksDB is in use; resync, replay, or restore from snapshot to switch back to chainbase");
```

Comment:
Where/when is backing_store_type::CHAINBASE state saved in the chainbase? Should a statement like db.modify(db.get<kv_db_config_object>(), [](auto& cfg) { cfg.backing_store = backing_store_type::CHAINBASE; }); appear somewhere?

Reply:
Where would you suggest this statement should appear?

Reply:
Whenever configuration is changed from RocksDB to Chainbase, I think that should be reflected in the kv_db_config_object in the database. Maybe this kv_db_config_object's name should be changed too as it contains
information not only for kv.

Reply:
It's not possible to switch from RocksDB to Chainbase, because, unlike the opposite case - Chainbase to RocksDB, there is no check for whether RocksDB was actually used. Once nodeos has been switched to RocksDB, that change is permanent, and a replay or restore from snapshot would be required to switch back to Chainbase.

Reply:
I see. Thanks. If backing_store is Chainbase, kv_db_config_object will be created to its default value backing_store_type::Chainbase.

Reply:
Yes.